SwiftServer/Sources/HTTPResponse/HTTPRequestParser.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+enum ParsingErrors: Error {
+    case BadRequest
+}
+
+public class StartLine {
+    public var httpMethod: String
+    public var target: String
+    public var httpVersion: String
+
+    init(startLine: String) throws {
+        var startLineArray = startLine.components(separatedBy: " ")
+        if startLineArray.count != 3 {
+            throw ParsingErrors.BadRequest
+        }
+        httpMethod = startLineArray[0]
+        target = startLineArray[1]
+        httpVersion = startLineArray[2]
+    }
+}
+
+public class HTTPRequestParser {
+    public var startLine: StartLine!
```

Comment:
I think you should aim to keep the names as close as possible to the W3 standard, in this case `requestLine`:
```5.1 Request-Line

The Request-Line begins with a method token, followed by the Request-URI and the protocol version, and ending with CRLF. The elements are separated by SP characters. No CR or LF is allowed except in the final CRLF sequence.

        Request-Line   = Method SP Request-URI SP HTTP-Version CRLF
```
It will be a lot easier to read and the code will practically write itself :)